types/node/events.d.ts

```diff
@@ -639,3 +676,125 @@ declare module 'node:events' {
     import events = require('events');
     export = events;
 }
+
+// NB: The Event / EventTarget / EventListener implementations below were copied
+// from lib.dom.d.ts, then edited to reflect Node's documentation at
+// https://nodejs.org/api/events.html#class-eventtarget.
+// Please read that link to understand important implementation differences.
+
+// For now, these Events are contained in a temporary module to avoid conflicts
+// with their DOM versions in projects that include both `lib.dom.d.ts` and `@types/node`.
+// Library consumers should *not* import from this fictional module!
+declare module '__dom-events' {
```

Comment:
What else uses this aside from the commented section on line 42? Is it still needed? 

It also doesn't seem necessary to introduce a type-only pseudo-module when these definitions could be moved inside the `events` module and just not exported (i.e., add an `export` to each declaration in `events` along with an `export {};` to indicate that declarations in the module aren't exported by default.

Reply:
Not gonna lie, I only learned about `export {}` about a week ago.  As it turns out though, that isn't even necessary, because the `events` module uses a default-assignment export (`export = EventEmitter`) so everything else is already not exported.

It occurred to me though, the DOM-like Event(Target) is only used globally, never exported from any module, and contains no references to anything from the `event` module, so there's no reason for that code to live in this file.  I moved it to its own file in the latest commit.